### DISCUSSION TITLE
Hilfe bei der Installation - HMIP Geräte werden nicht erkannt

### DISCUSSION THREAD

**Discussion by mario81k** [SukramJ/aiohomematic#792] [Q&A]
Hallo Zusammen,
.i.ch versuche seit Tagen meine CCU3 in HA zu integrieren und habe die Installation auch nach Anleitung durchgeführt. Die CCU3 wird auch erkannt - leider werden aber auch nach mehreren Neustarts von HA die Geräte nicht übertragen. Ich bin wirklich am verzweifeln und hoffe das mir hier jemand helfen kann.
Zur Installation:
HomeAssistant läuft bei mir in einer VM auf dem QNAP-NAS unbd ist über die interne IP 192.168.2.207 erreichbar:
hier mal die Netzübersicht - hier wundert mich das bei der VM die IP nicht angezeigt wird:
![image](https://user-images.githubusercontent.com/121456551/209933906-324d6636-4877-45f5-bd93-f35989565e2f.png)
Home Assistant 2022.12.8
Supervisor 2022.12.1
Operating System 9.4
Frontend 20221213.1 - latest
die CCU3 ist im selben Subnetz unter 192.168.2.134 erreichbar, die Firewall habe ich zu Testzwecken mal komplett geöffnet:
![image](https://user-images.githubusercontent.com/121456551/209933720-8d760be4-077e-421f-a35a-f39d061eae24.png)
Die CCU3 wird in HA wie folgt angezeigt:
![image](https://user-images.githubusercontent.com/121456551/209934150-a71e307c-9353-4a80-9d79-76226e403176.png)
Die Konfig sieht wie folgt aus:
![image](https://user-images.githubusercontent.com/121456551/209934248-57aae99e-a70c-4b37-929d-0ebaab0612c6.png)
![image](https://user-images.githubusercontent.com/121456551/209934328-26bd2e03-6fdb-40a8-afbd-78fe463facdf.png)
Hier auch die Diagnoseinfos:
[config_entry-homematicip_local-a291e3c2cdf2e854c1e3e8f3b9143103.json (1).txt](https://github.com/danielperna84/hahomematic/files/10318741/config_entry-homematicip_local-a291e3c2cdf2e854c1e3e8f3b9143103.json.1.txt)
Ich hoffe wirklich das mit hier jemand weiterhelfen kann. Bin Neuling auf dem Gebiet und gerade ziemlich am verzweifeln.
Vorab vielen Dank!
Gruß Mario
PS: Sollte ich wichtige Infos vergessen haben gerne einfach nachfragen!

**Comment by SukramJ** [comment]
Wenn ich das richtig gelesen habe verwendest du HAOS und keine standalone docker Installation. Richtig?

Aus der [Doku](https://github.com/danielperna84/custom_homematic#firewall-and-required-ports):
> This integration starts a local XmLRPC server within HA, which automatically selects a free port or uses the optionally defined callback port. This means that the CCU must be able to start a new connection to the system running HA and to the port. So check the firewall of the system running HA (host/VM) to allow communication from the CCU.

Irgendwelche Firewallregeln auf dein HA VM gesetzt?

**Comment by slackstarr** [comment]
Hi!

Ich stand gestern vor einer ähnlichen Herausforderung und habe nach langer Recherche eine Lösung gefunden ([Link zur Website](https://forum.simon42.com/home-assistant-allgemein/raspberrymatic-addin-zugriff-auf-raspberrymatic-extern/#post-178)). 

Kurz zusammengefasst: 

- Im CCU3-Webgui (Einstellungen > Systemsteuerung) den Sicherheitsassistenten aufrufen und die Sicherheitseinstellungen per Express auf „locker“ einstellen. -> Dialog schließen
- Unter Sicherheit (Einstellungen > Systemsteuerung -> Sicherheit) im Bereich Authentifizierung die Checkbox deaktivieren.
- FW-Richtlinie: Ports offen
- Die weiteren 3 Pulldown-Menüs sehen bei mir so aus: eingeschränkt, eingeschränkt, Vollzugriff
- Folgende Ports in der Firewall freigeben (sind bewusst auch TLS/SSL-Ports drin, da ich daran noch arbeite): 2001;2010;2000;9292;9293;1999;80;42001;42010;42000;49292;443
- IP-Adressen für den einschränkten Zugriff freigeben (das X durch deine Were ersetzen): 192.168.X.1;192.168.X.0/16;fc00::/7
- Und jetzt folgt die absolut bescheuertste Lösung: neuen Benutzer ohne Admin-Rechte anlegen (Wichtig: Passwort ohne Sonderzeichen!). Mit diesem Benutzer meldest du dich dann im Home Assistant Add-on an. Wenn dann alle Geräte, etc. gefunden wurden, kannst du den Benutzer zu einem Admin machen.

Ich hoffe, dass ich dir helfen konnte!

**Reply by SukramJ** [reply]
Haltet Euch an die [Anleitung](https://github.com/danielperna84/custom_homematic).
- es Wird ein Benutzer mit Admin rechten benötigt
- Passwort ohne Sonderzeichen
- Authentifizierung aktivieren.
- Firewall öffnen
- **Die CCU braucht Zugriff auf die HA Instanz**

**Reply by slackstarr** [reply]
Wie freundlich! Gestern mich noch bitten, meine Lösung zu beschreiben und heute sich darüber lustig machen. Besten Dank!

**Reply by SukramJ** [reply]
Tut mit Leid ich wollte nicht unfreundlich sein, aber lies dein Anleitung doch mal durch.
Erst einen Benutzer anlegen, und erst später Admin Rechte vergeben? 
Authentifizierung deaktivieren?
Firewall auf der CCU mit IPv6 Rechten?
Das ist leider sehr fragwürdig, und wird die meisten Benutzer mehr als nur irritieren.

Ich stelle leider fest das die Benutzer eher Videos und die dazugehörigen Kommentare lesen, statt sich einmal mit der zur Verfügung gestellten [Doku](https://github.com/danielperna84/custom_homematic) auseinanderzusetzen. Die ist bei weitem nicht perfekt, aber wir versuchen da alle relevanten Informationen zu sammeln. Wer sich an der Doku stört kann sich natürlich auch konstruktiv beteiligen.

Ich weiss wie ich die integration einrichte, denn das hab ich bestimmt schon 100 mal gemacht. 
Ich Frage nach, wenn hier Diskussionen/Issues aufgemacht werde und mit einem "Danke bei mir klappt's" für die Benutzer beendet ist, ohne das ursächliche Problem oder die Lösung zu nennen. Bei sinnvollen Lösungen sollen die Informationen dann wieder in die Doku einfliessen.

**Reply by slackstarr** [reply]
Hey SukramJ,

vielen Dank für deine Antwort. 

Wie ich in meinem anderen Post gestern schrieb, habe ich mich tatsächlich sehr lange mit der Doku auseinander gesetzt und bin leider nicht weiter gekommen. Ich schätze die Zeit, die Leute wie du, in solche Projekte steckt sehr und möchte niemandes Zeit verschwenden. Insofern war meine Antwort von heute ein Versuch meine Erfahrungen zu beschreiben - mea culpa, wenn die Beschreibung verwirrend bzw. schlicht falsch zu sein scheint. Ich werde mir eure Doku noch Mal genau durchlesen und gerne Feedback geben, an welcher Stelle etwas unverständlich bzw. nicht nachvollziehbar für mich war.

 Ich wünsche dir einen guten Rutsch ins neue Jahr!

**Reply by SukramJ** [reply]
Danke wünsche ich dir auch.

Feedback zur Doku gerne auch als Pull request

**Reply by mario81k** [reply]
Hallo Zusammen,
ich habe es leider erst heute geschafft mich nochmal an das Thema zu setzen. Dank eurer Hilfe mit Erfolg. Es lag wirklich an den Sonderzeichen im Passwort - sorry, da hätte ich auch selbst drauf kommen könnend
Die Integration läuft jetzt perfekt. Vielen Dank für eure Unterstützung!
Ich wünsche euch einen guten Rutsch in ein gesundes 2023!!!

**Comment by Margas80** [comment]
Hi, Ich habe das selbe Problem, bin recht neu in HA.
Der hier beschrieben Lösungsansatz hat mit nicht geholfen.
Folgende Fehlermeldung im Log kommt.
![Homaric Error_2](https://user-images.githubusercontent.com/118937205/210242605-0e9799bb-f8a0-47ff-a596-883f900748ba.JPG)
Wenn ich die Einstellungen direkt in der config.yaml mache bekomme ich alle Geräte als Entitäten, außer der neuen PSM2 Steckdose. 
Deshalb wollte ich auf diese Intergration wechseln, damit es sauber dargestellt wird.
Komme hier nicht weiter.
Die Integration findet die CCU3 aber es werden halt keine Geräte (Entitäten) erkannt.
Bei jedem Neustart sagt er wieder er hat die CCU3 entdeckt und ich soll sie konfigurieren. Das habe ich aber schon zig mal versucht.
Die Ports auf der CCU3 sind offen und habe zwar einen Admin User der für die Anmeldung genutzt wird aber das Kennwort ist bis auf zwei Ausrufezeichen(innerhalb des Wortes), Sonderzeichenfrei.
Bräuchte hier echt support von jemanden.

**Reply by Margas80** [reply]
Vergessen zu sagen.
HA läuft bei mir auf einem PI4 und CCU3 ist separat.

**Reply by SukramJ** [reply]
Deaktiviere die "Automatische Umleitung auf HTTPS" in der CCU oder aktiver TLS in der Integration.

**Reply by Margas80** [reply]
Danke, das wars
